modeshape-jcr/src/main/java/org/modeshape/jcr/txn/Transactions.java

```diff
@@ -384,11 +384,12 @@ public void commit()
 
     protected class NestableThreadLocalTransaction extends TraceableSimpleTransaction {
         private AtomicInteger nestedLevel = new AtomicInteger(0);
-        private ThreadLocal<? extends Transaction> txReference;
+        private ThreadLocal<NestableThreadLocalTransaction> txHolder;
```

Comment:
`NestableThreadLocalTransaction`s are only used when there is no external transaction - i.e. when ModeShape starts & commits a transaction during `session.save()`. If that is the case, these type of transactions are "internal" to Modeshape and not visible on the outside, hence they cannot be context-switched.

Context switching on the other hand is specific/possible only to user-transactions (i.e. `SynchronizedTransaction` instances) which are not held anymore in `ThreadLocal` variables.